File: LeetCode-Solutions/0001-two-sum/solution.cpp
#define deb(x) cerr << #x << ": " << x << '\n'
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = (int)nums.size();
        int i = 0, j = n-1;
        vector<int> ans;
        vector<pair<int,int>> nums2(n);
        for(int idx = 0; idx < n; ++idx){
            nums2[idx] = make_pair(nums[idx], idx);
        }
        sort(nums2.begin(), nums2.end());
        while(i < j){
            int sum = nums2[i].first + nums2[j].first;
            if(sum> target)
                {j--;}
            else if (sum < target)
                {i++;}
            else
                {ans.push_back(nums2[i].second);ans.push_back(nums2[j].second); break;}
        }
        return ans;
    }
};

File: LeetCode-Solutions/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] answer = new int[2];
        int index = 0;
        int index2 = 0;

        while(index != nums.length){
            int diff = target - nums[index];
            for(int i = index+1; i < nums.length; i++){
                if(nums[i] == diff){
                    index2 = i;
                    break;
                }
            }
            if(index2 != 0){
                break;
            } else {
                index++;
            }
        }

        if(index2 > index){
            answer[0] = index;
            answer[1] = index2;
            return answer;
        }

        int[] noanswer = new int[0];
        return noanswer;
    }
}
